src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -104,6 +104,13 @@ protected TestClass createTestClass(Class<?> testClass) {
     protected abstract Description describeChild(T child);
 
     /**
+     * Fires test ignore through {@code notifier} for child corresponding to
+     * {@code child}, which can be assumed to be an element of the list returned
+     * by {@link ParentRunner#getChildren()}.
+     */
+    protected abstract void fireIgnoreChild(T child, RunNotifier notifier);
```

Comment:
Yes, I see, you are right. My first suggestion was to move the abstract method to parent class Runner, but this could be a bad design, because fireIgnoreChild() and runChild() will then be declared in different classes.
I am now thinking of removing fireIgnoreChild(), because runChild() is enough for my purpose.